### PR TITLE
Reuse more templates in our build

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Make the generate-offsets and workloads jobs based on the global-build-job template.

Make the publish-signed-artifacts job based on the Arcade template and use more 1ES pipeline features (as it's internal only and doesn't need to run in a non-1ES context).

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
Official Build Run: https://dnceng.visualstudio.com/internal/_build/results?buildId=2432198&view=results

**Comment by ivdiazsa** [comment]
Seems like all failures are unrelated. Would we benefit from also running the outerloop pipeline?

**Comment by jkoritzinsky** [comment]
This change only affects the runtime and runtime-official pipelines, so there's no need to run any outerloop pipelines.

File: eng/pipelines/runtime-official.yml [hunk 1]
```diff
@@ -381,11 +381,13 @@ extends:
                 parameters:
                   name: MonoRuntimePacks
 
+
       # Build Mono AOT offset headers once, for consumption elsewhere
+      # Only when mono changed
```
Comment (akoeplinger) [inline eng/pipelines/runtime-official.yml:386]: we don't use the evaluate-paths logic in runtime-official
```suggestion
```

File: eng/pipelines/runtime-official.yml [hunk 2]
```diff
@@ -394,9 +396,37 @@ extends:
...
+                displayName: 'Publish Mono AOT offsets'
+                condition: succeeded()
+                targetPath: '$(Build.SourcesDirectory)/artifacts/obj/mono/offsetfiles'
+                artifactName: 'Mono_Offsets_$(osGroup)$(osSubGroup)'
+            # needed by crossaot
+            condition: >-
+              or(
+                eq(stageDependencies.EvaluatePaths.evaluate_paths.outputs['SetPathVars_mono_excluding_wasm.containsChange'], true),
+                eq(stageDependencies.EvaluatePaths.evaluate_paths.outputs['SetPathVars_installer.containsChange'], true),
+                eq(variables['isRollingBuild'], true))
```
Comment (akoeplinger) [inline eng/pipelines/runtime-official.yml:429]:
```suggestion
```

**Review by akoeplinger — APPROVED** [review]
One comment/change, LGTM otherwise

**Pull request merged**